honeybee_radiance_command/dctimestep.py

```diff
@@ -0,0 +1,323 @@
+        day_coef_matrix: Path to the input daylight coefficient matrix  file, generated
+            by rfluxmtx.
+        sun_coef_matrix: Path to the input sun_coefficient matrix file, generated by
+            rcontrib or rfluxmtx.
+        view_matrix: Path to the input view matrix, created by rfluxmtx, for three-phase,
+            four-phase, three-phase-direct and  four-phase-direct simulations.
+        t_matrix: Path to the input t (transmission matrix) BSDF file in xml format.
+        daylight_matrix: Path to the input daylight matrix, created by rfluxmtx, for
+            three-phase, four-phase, three-phase-direct and  four-phase-direct +
+            simulations.
+        facade_matrix: Path to the input facade matrix, created by rfluxmtx, for
+            four-phase and four-phase-direct simulations.
+        is_direct_sun_calc: Set to True if the simulation is a direct sun calculation.
+        is_daylight_coef_calc: Set to True if the simulation is a daylight coefficient
+            calculation.
+        is_three_phase_calc: Set to True if the simulation is a three phase simulation.
+        is_four_phase_calc: Set to True if the simulation is a four phase simulation.
+
+    Properties:
+        * options
+        * output
+        * sky_vector
+        * sun_vector
+        * day_coef_matrix
+        * sun_coef_matrix
+        * facade_matrix
+        * view_matrix
+        * daylight_matrix
+        * t_matrix
+        * is_direct_sun_calc
+        * is_daylight_coef_calc
+        * is_three_phase_calc
+        * is_four_phase_calc
+
+    """
+
+    __slots__ = ('_sky_vector', '_day_coef_matrix', '_view_matrix', '_daylight_matrix',
+                 '_t_matrix', '_sun_vector', '_sun_coef_matrix', '_facade_matrix',
+                 '_is_direct_sun_calc', '_is_daylight_coef_calc', '_is_three_phase_calc',
+                 '_is_four_phase_calc')
+
+    def __init__(self, options=None, output=None, sky_vector=None, sun_vector=None,
+                 day_coef_matrix=None, sun_coef_matrix=None, view_matrix=None,
+                 t_matrix=None, daylight_matrix=None, facade_matrix=None,
+                 is_direct_sun_calc=False, is_daylight_coef_calc=False,
+                 is_three_phase_calc=False, is_four_phase_calc=False):
+        """Initialize command"""
+        Command.__init__(self, output=output)
+        self.options = options
+        self.sky_vector = sky_vector
+        self.sun_vector = sun_vector
+        self.day_coef_matrix = day_coef_matrix
+        self.sun_coef_matrix = sun_coef_matrix
+        self.view_matrix = view_matrix
+        self.t_matrix = t_matrix
+        self.daylight_matrix = daylight_matrix
+        self.facade_matrix = facade_matrix
+        self.is_direct_sun_calc = is_direct_sun_calc
+        self.is_daylight_coef_calc = is_daylight_coef_calc
+        self.is_three_phase_calc = is_three_phase_calc
+        self.is_four_phase_calc = is_four_phase_calc
```

Comment:
I think you should not expose the setter for these values and manage them inside your code. A typical user will not know any of the details here on how to manage them.

I would also add a note to the class that one should use the classmethods to initiate this class.

Reply:
Okay